src/App.tsx

```diff
@@ -1,14 +1,73 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+
+import { Todo } from './types/Todo';
+
+type Params = {
+  query: string;
+  filter: string;
+};
+
+function filterTodos(list: Todo[], params: Params) {
+  const { query, filter } = params;
+  let result: Todo[] = [];
+
+  if (filter) {
+    switch (filter) {
+      case 'all':
+        result = list;
+        break;
+
+      case 'active':
+        result = list.filter((todo: Todo) => !todo.completed);
+        break;
+
+      case 'completed':
+        result = list.filter((todo: Todo) => todo.completed);
+        break;
+
+      default:
+        result = list;
+        break;
```

Comment:
The `filter` state is initialized with 'All', but the switch cases use lowercase 'all'. This inconsistency can lead to the default case being executed unintentionally. Ensure the filter values are consistent.